.circleci/config.yml

```diff
@@ -0,0 +1,52 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/app-prototype
+    docker:
+      - image: circleci/ruby:2.4.2-node-browsers
+        environment:
+          PGHOST: 127.0.0.1
+          PGUSER: postgres
+          RAILS_ENV: test
+      - image: circleci/postgres:9.6.5-alpine
+        environment:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: app_prototype_test
+          POSTGRES_PASSWORD: ""
+    steps:
+      - checkout
+
+      # Restore bundle cache
+      - type: cache-restore
+        key: bundle-{{ arch }}
```

Comment:
I think this should also include `{{ checksum "Gemfile.lock" }}`? (Also on line 29)

https://circleci.com/docs/2.0/language-ruby/#sample-configuration

Reply:
Yeah, that's what the docs say, but it forces a full fetch on *any* lockfile change. That doesn't seem necessary to me. This achieves the same end state (only the gems in the lockfile are present in the vendor dir) without incurring the huge increase in build times whenever the lockfile changes. Right?

Reply:
Hmm, I think you're right. Since you are also doing `bundle clean` this should be better and faster.

Reply:
Fair point. Anecdotally it seemed like CCI2's could hold onto at least 2 sets of gems, so someone could install new gems in one branch without affecting the other builds.

Reply:
I was just looking into this yesterday. It looks like Circle supports setting multiple, cascading cache keys:

https://circleci.com/docs/2.0/caching/#full-example-of-saving-and-restoring-cache

I believe it tries the longest, most specific keys first, then falls back the the less specific key prefixes. Theoretically, that should prevent branches with different gems from interfering with the caches on other branches.

Reply:
Neat. Though, thinking through it, what kind of interference could happen with the current caching strategy?

Assume we have 2 branches: `apples` and `oranges`. A build on `oranges` might remove gem `lib1` every time it runs, while a build on `apples` may add `lib1` on every build. It's a tiny bummer to see `lib1` removed on `oranges` builds, and added on `apples`.

It shouldn't change the run time that much, though that's really a function of just how many gems have changed between branches.

Each build -- due to the `bundle clean` -- ends up with _only_ the gems specified by the lockfile, so there shouldn't be any weird gem related failures.

Or am i missing something... something more dire?

@subakva how would you use the cascading caches to resolve this issue?

Reply:
Oh, I see it now. Hmmmm... I'll give it a try.